pkg/postgres/postgres.go

```diff
@@ -0,0 +1,132 @@
+package postgres
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgx"
+	logger "github.com/phil-inc/plog-ng/pkg/core"
+)
+
+var pool *pgx.ConnPool
+var config *Config
+
+type Config struct {
+	URL         string
+	DBName      string
+	SSLMode     string
+	SSLRootCert string
+}
+
+type Tx struct {
+	pgx.Tx
+}
+
+type Rows struct {
+	pgx.Rows
+}
+
+var ErrNoRows = pgx.ErrNoRows
+
+func connectPostgres(connConfig *Config) (*pgx.ConnPool, error) {
+	connURL := fmt.Sprintf("%s/%s?sslmode=%s", connConfig.URL, connConfig.DBName, connConfig.SSLMode)
+
+	if connConfig.SSLMode != "disable" && strings.Trim(connConfig.SSLRootCert, " ") != "" {
+		connURL = fmt.Sprintf("%s&sslrootcert=%s", connURL, connConfig.SSLRootCert)
+	}
+
+	connectionConfig, err := pgx.ParseURI(connURL)
+	if err != nil {
+		return nil, err
+	}
+
+	maxConnections := 50
+	timeOut := 5 * time.Minute
```

Comment:
5 mins is a long time isn't it?

Reply:
Since this is the max wait time when all connections are busy,  in the case of jobs we would ideally like to wait for a prolonged time in case all the connections are busy so we have been using 5 mins. What do you suggest the time should be?